request_handler.go

```
@@ -199,97 +184,129 @@ func createHandler(w http.ResponseWriter, r *http.Request, title string){
+	// Encode data to Json
 	_, err = http.Post("http://localhost:8080/v1/accounts/", "application/json", payload)
 	if err != nil {
 		log.Println(err)
 	}
-	err = login(w, pageinfo.Username, pageinfo.Password)
+	// Send the request to create a new account
+
+	user := LogInfo{}
+	user.Username = pageinfo.Username
+	user.Password = pageinfo.Password
+	userData, err := json.Marshal(user)
+	userString := string(userData)
+	payload = strings.NewReader(userString)
+	response, err := http.Post("http://localhost:8080/v1/sessions/", "application/json", payload)
+	if err != nil {
+		log.Println(err)
+	}
+	// Log in with newly created account information
+	// Get token from login response from backend
+	token := response.Header.Get("Set-Cookie")
+	Cookie := http.Cookie{Name:"name-cookie",
+		Value: token,
+		Path: "/",
+		HttpOnly:true}
+	http.SetCookie(w, &Cookie)
+	// Set the cookie to the browser
 	if err != nil {
 		log.Println(err)
 	}
+	// After log in, redirect to the personal private page
 	http.Redirect(w, r, "/privatePage/", http.StatusFound)
 }
 
 // Handle the login page
-
 func homeHandler(w http.ResponseWriter, r *http.Request, title string) {
 	p := Profile{}
 	renderTemplate(w, "login", &p)
 }
 
-// Do the login job and set the token
 
-func login(w http.ResponseWriter, username string, password string) error {
+//
+func loginHandler(w http.ResponseWriter, r *http.Request, title string) {
+	// Get user login in information
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	// Encode the log in data to the json format payload
 	user := LogInfo{}
 	user.Username = username
 	user.Password = password
 	userData, err := json.Marshal(user)
 	userString := string(userData)
 	payload := strings.NewReader(userString)
 	response, err := http.Post("http://localhost:8080/v1/sessions/", "application/json", payload)
+	if response.StatusCode == 401 {
+		http.Redirect(w, r, "/loginError/", http.StatusFound)
+	}
 	if err != nil {
 		log.Println(err)
```

It shouldn't just ignore error.